server/src/main/java/org/elasticsearch/search/aggregations/bucket/filter/FiltersAggregator.java

```diff
@@ -135,8 +135,22 @@ public static FiltersAggregator build(
         CardinalityUpperBound cardinality,
         Map<String, Object> metadata
     ) throws IOException {
-        if (canUseFilterByFilter(parent, factories, otherBucketKey)) {
-            return buildFilterByFilter(name, factories, filters, keyed, otherBucketKey, context, parent, cardinality, metadata);
+        if (canUseFilterByFilter(parent, otherBucketKey)) {
+            FilterByFilter filterByFilter = buildFilterByFilter(
+                name,
+                factories,
+                filters,
+                keyed,
+                otherBucketKey,
+                context,
+                parent,
+                cardinality,
+                metadata
+            );
+            if (false == filterByFilter.scoreMode().needsScores()) {
+                // Filter by filter won't produce the correct results if the sub-aggregators need scores
+                return filterByFilter;
```

Comment:
This bit is pretty sad! But it isn't worth trying to figure this stuff out now. Maybe not never. I wish it weren't so janky to stop it.

Reply:
Yeah, that scares me quite a bit. Can you add a comment explaining that this information is not readily available until we actually build aggregations. I feel like factories should have enough information to actually know this stuff.

Reply:
They might be able to tell, but it'd be a ton of plumbing. I looked at it and though "I think it'd be easier to make score work here". But I didn't want to do either one right now..... I'll add the comment!

Reply:
Yeah, I looked at it as well, and it didn't look practical to fix right now. My comment was more like a mental note to add this to the wish list if we ever get to refactor them.